File: campus-card-wechat-api/src/main/java/com/campus/card/wechat/model/Feedback.java
package com.campus.card.wechat.model;

import javax.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "feedback")
public class Feedback {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "child_id")
    private Long childId;

    @Column(name = "category", length = 32)
    private String category; // BUG, SUGGESTION, OTHER

    @Column(name = "content", length = 1024)
    private String content;

    @Column(name = "contact", length = 128)
    private String contact;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "status", length = 20)
    private String status; // NEW, HANDLED

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public Long getChildId() { return childId; }
    public void setChildId(Long childId) { this.childId = childId; }
    public String getCategory() { return category; }
    public void setCategory(String category) { this.category = category; }
    public String getContent() { return content; }
    public void setContent(String content) { this.content = content; }
    public String getContact() { return contact; }
    public void setContact(String contact) { this.contact = contact; }
    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }
    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }
}

File: campus-card-wechat-web/src/api/modules/feedback.ts
import http from '../request'

export interface FeedbackItem {
  id: number
  childId: number
  category?: string
  content: string
  contact?: string
  status?: string
  createdAt: string
}

export function submitFeedback(data: { childId?: number; category?: string; content: string; contact?: string }) {
  return http.post<FeedbackItem>('/api/v1/feedback/submit', data)
}

export function getFeedbackList(childId?: number) {
  return http.get<FeedbackItem[]>('/api/v1/feedback/list', { params: { childId } })
}